src/append/file.rs

```diff
@@ -28,9 +29,10 @@ impl fmt::Debug for FileAppender {
 }
 
 impl Append for FileAppender {
-    fn append(&mut self, record: &LogRecord) -> Result<(), Box<Error>> {
-        try!(self.encoder.encode(&mut self.file, record));
-        try!(self.file.flush());
+    fn append(&self, record: &LogRecord) -> Result<(), Box<Error>> {
+        let mut file = self.file.lock().expect("encode doesn't panic; no poison");
```

Comment:
I think we can actually just ignore the poision with `lock().unwrap_or_else(|e| e.into_inner())`, since the mutex only protects the writer which doesn't have any panic safety issues.